### DISCUSSION TITLE
how to use the CameraCalibration node

### DISCUSSION THREAD

**Discussion by albmef** [alicevision/Meshroom#1780] [Q&A]
Hi

i run meshroom 2021.1.0 on kubuntu 22 LTS

I managed to calibrate my camera sensor using the CameraCalibration node, with a CHESSBOARD 9x6 pattern
my camera is a Lumix G80 DSLR with stock 14-140 lens

here is the node configuration 

Program called with the following parameters:
 * calibGridSize = 10
 * debugRejectedImgFolder = ""
 * debugSelectedImgFolder = ""
 * input =  Unknown Type "N5boost10filesystem4pathE"
 * maxCalibFrames = 100
 * maxFrames = 0
 * maxTotalAvgErr = 0.1
 * minInputFrames = 10
 * nbDistortionCoef = 3
 * output = "/home/alban/Bureau/G80cal/20220923/MeshroomCache/CameraCalibration/52b9364e9ca0e7c5f94f6d02bc6dab27657bddde/cameraCalibration.cal"
 * pattern =  Unknown Type "N11aliceVision11calibration7PatternE"
 * size =  Unknown Type "St6vectorImSaImEE"
 * squareSize = 40
 

here are the results in cameraCalibration.cal.txt:

4592
3448
3764.67
2293.82
1757.03
-0.129042
0.892266
-4.24153

here are the results in cameraCalibration.cal:

calibration_time: "Fri Sep 23 14:21:32 2022\n"
image_width: 4592
image_height: 3448
board_width: 9
board_height: 6
square_size: 40.

flags: 14344
camera_matrix: !!opencv-matrix
   rows: 3
   cols: 3
   dt: d
   data: [ 3.7556532499402060e+03, 0., 2.2938236005965464e+03, 0.,
       3.7736798001841366e+03, 1.7570263683244921e+03, 0., 0., 1. ]
distortion_coefficients: !!opencv-matrix
   rows: 5
   cols: 1
   dt: d
   data: [ -1.2904202252322350e-01, 8.9226570899180824e-01, 0., 0.,
       -4.2415293307904571e+00 ]
avg_reprojection_error: 1.0964765025260934e+00

1> someone could kindly explain me how to parse these results and use them in CameraInit node ? 
this way i could use BMP images extracted from video files that not contains EXIF data,
or use a smart phone camera whose sensor is not referenced in sensorDB

2> what is the difference between "square size" and "calib grid size" parameters ?

best regads,
Alban

**Comment by natowi** [comment]
The updated documentation should help :)
https://meshroom-manual.readthedocs.io/en/latest/feature-documentation/nodes/CameraCalibration.html

**Reply by albmef** [reply]
thanks !
